README.md

```diff
@@ -51,8 +51,27 @@ You need the following tools to build the library with the provided makefiles:
 
 * GNU Make or a build tool that CMake supports.
 * A C99 toolchain (compiler, linker, archiver). We actively test with GCC 5.4, Clang 3.8, IAR8 and Visual Studio 2013. More recent versions should work. Slightly older versions may work.
-* Python 3 to generate the test code.
-* Perl to run the tests.
+* Python 3 to generate the test code, and to generate sample programs in the development branch.
+* Perl to run the tests, and to generate some source files in the development branch.
+
+### Generated source files in the development branch
+
+The source code of Mbed TLS includes some files that are automatically generated by scripts and whose content depends only on the Mbed TLS source, not on the platform or on the library configuration. These files are not included in the development branch of Mbed TLS, but the generated files are included in official releases. This section explains how to generate the missing files in the development branch.
+
+The following tools are required:
+
+* Perl, for some library source files and for Visual Studio build files.
+* Python 3, for some sample programs and test data.
+* A C compiler for the host platform, for some test data.
+
+If you are cross-compiling, you must set the `CC` environment variable to a C compiler for the host platform when generating the configuration-independent files.
+
+Any of the following methods are available to generate the configuration-independent files:
+
+* If not cross-compiling, running `make` with any target, or just `make`, will automatically generate required files.
+* Run `make generated_files` to generate all the configuration-independent files.
+* On Unix/POSIX systems, run `tests/scripts/check-generated-files.sh -u` to generate all the configuration-independent files.
```

Comment:
I don't think that works without [the first commit of 4514](https://github.com/ARMmbed/mbedtls/pull/4514/commits/7f14c433513409d993f4a1dcf6d6912a0fcd9707) - which you might want to cherry-pick here for consistency with the documentation.

Reply:
Good find, but I don't like the way you do it: it causes `check-generated-files -u` to always print messages about the files being out of date, even when there are no changes. So I did it slightly differently.

I didn't fix the annoyance that `check-generated-files -u` always updates the mtime on the files, even if they end up with the same content. I consider this out of scope, and less important now that you can use `make` if you care about timestamps.